gsoc_summary.md

```diff
@@ -0,0 +1,82 @@
+	
+  Originally, the environment for this process was defined inside one of the main services of the bot, not only that, but the updates also occurred inside the current computation of the graph. 
+
+<p align="center">
+  <img src="https://github.com/viniciusdc/viniciusdc.github.io/blob/viniciusdc-patch-3/img/old_eco.png" />
+</p>
+ 
+So, the fundamental idea for this project  is to move the data structures used to update the packages to another repo with a separate process, not interfering directly with the graph and the others bot microservices, for instance the Migrations. We should than have a new environment for this process, in a much cleaner way
+
+<p align="center">
+  <img src="https://github.com/viniciusdc/viniciusdc.github.io/blob/viniciusdc-patch-3/img/new_eco.png" />
+</p>
+
+As we can see, the idea is quite valuable and simplistic, even though not only there is a bigger change into the update versions code, as now should not bump any information inside the graph structure while loading the versions but also, as we actually are performing these evaluation outside the main process asynchronously, we now need another microprocesses to get this data and insert back to the graph to perform the other process.
+
+
+### The proposed updates
+  - Separate/upgrade the code that search for version updates  from the graph structure;
+  - Create new a process on Circle CI to manage the new versions and bump then at a new folder/repository at GitHub;
+  - Adapt the ‘make graph’ process to load the versions information from the new location and insert then at the graph;
+  - Adapt the tests codes for the new scripts and functions;
+  - Optimize minor parts of the new update versions code.
+  * ## Optional
+  	- As another improvement for the bot usage, a profiler decorator (with context manager) was implemented. [extra](https://github.com/regro/cf-scripts/pull/1131#pullrequestreview-474906393)
+
+### The new code workflow
+   Before starting with the modification an improved explanation about the update versions process shall be given. 
+   
+   First we load the graph (which is the fundamental structure for accommodating information about the feedstocks), once loaded, we search for every fedstock the information regarding its versions at their sources (which can be PyPi, ROS, NPM and so on). Then a special function called `get_latest_version` is responsible for the requisition of the new versions from the sources and `update_upstream_version` updates the information at the graph with the requested ones, this process comprehends the 'update version process', wich in this case is attached to the graph main structure.
```

Comment:
I think that `process'` is accepted now but I recall learning a long time ago that it should br `process's`. I'll let those who have English as a first language to comment on this since I'm probably outdated in the s and ' rules.